contribute/design/cstor-operator/specs.md

```diff
@@ -0,0 +1,423 @@
+      // For example -- cStor mirrored group can have 2 block devices only.
+
+      // Name is the name of the group.
+      // An user should put a group name and it should be unique for this
+      // node spec.
+    - name: group1
+      blockDevices:
+        // Name of the block devices can be obtained using `kubectl get bd`
+      - name: sparse-3c1fc7491f9e4cf50053730740647318
+        // id can be specified by an user and it must be unique for this node
+        // spec.
+        // If not provided by user -- it will be system generated and put.
+        id: bd1
+
+```
+
+2.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-striped
+spec:
+  nodes:
+  - name: gke-cstor-it-default-pool-1
+    poolSpec:
+      poolType: striped
+    groups:
+    - name: group1
+      blockDevices:
+      - name: pool-1-bd-1
+        id: pool-1-bd1
+      - name: pool-1-bd-2
+        id: pool-1-bd2
+  - name: gke-cstor-it-default-pool-2
+    poolSpec:
+      poolType: striped
+    groups:
+    - name: group1
+      blockDevices:
+      - name: pool-2-bd-1
+        id: pool-2-bd1
+
+```
+
+3.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-mirrored
+spec:
+  nodes:
+
+  - name: gke-cstor-it-default-pool-1
+    poolSpec:
+      poolType: mirrored
+
+    groups:
+    - name: group1
+      blockDevices:
+      - name: pool-1-bd-1
+        id: pool-1-bd1
+
+      - name: pool-1-bd-2
+        id: pool-1-bd2
+
+    - name: group2
+      blockDevices:
+      - name: pool-1-bd-3
+        id: pool-1-bd3
+
+      - name: pool-1-bd-4
+        id: pool-1-bd4
+
+  - name: gke-cstor-it-default-pool-2
+    poolSpec:
+      poolType: striped
+
+    groups:
+    - name: group1
+      blockDevices:
+      - name: pool-2-bd-1
+        id: pool-2-bd1
+
+      - name: pool-2-bd-2
+        id: pool-2-bd2
+
+```
+4.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-hybrid
+spec:
+  nodes:
+
+  - name: gke-cstor-it-default-pool-1
+    poolSpec:
+      poolType: mirrored
+
+    groups:
+    - name: group1
+      blockDevices:
+      - name: pool-1-bd-1
+        id: pool-1-bd1
+
+      - name: pool-1-bd-2
+        id: pool-1-bd2
+
+    - name: group2
+      blockDevices:
+      - name: pool-1-bd-3
+        id: pool-1-bd3
+
+      - name: pool-1-bd-4
+        id: pool-1-bd4
+
+  - name: gke-cstor-it-default-pool-2
+    poolSpec:
+      poolType: striped
+
+    groups:
+    - name: group1
+      blockDevices:
+      - name: pool-2-bd-2
+        id: pool-2-bd2
+
+```
+
+Following is a sample auto CStorPoolCluster specs
+
+1.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-sparse-pool-test
+spec:
+  maxPools: 3
+  poolSpec:
+   poolType: raidz // Possible values : striped, mirrored, raidz or raidz2
+  name: cstor-sparse-pool-test
+  type: sparse // possible values : sparse or disk
```

Comment:
will it be good if we rename `type` to `diskType`. 
If this acts like a flag to determine a sparse vs disk & we believe this will be the case in future; then how about `isSparse` as the field name which is false by default?